akka-tools-persistence/src/main/scala/no/nextgentel/oss/akkatools/persistence/EnhancedPersistentActor.scala

```diff
@@ -160,10 +197,44 @@ abstract class EnhancedPersistentActor[E:ClassTag, Ex <: Exception : ClassTag]
       // Increment counter of events received since last NewDMGeneratingVersionEvent
       recoveredEventsCount_sinceLast_dmGeneratingVersion = recoveredEventsCount_sinceLast_dmGeneratingVersion + 1
       onReceiveRecover(event)
+    case offer: SnapshotOffer => {
+      if(!offer.isInstanceOf[FullSnapshotState]) {
+        throw new RuntimeException(s"Snapshot is not of the expected type ${FullSnapshotState.getClass.getName}) but was ${offer.getClass.getName}" )
+      }
+      val offerState = offer.snapshot.asInstanceOf[FullSnapshotState]
+      recoverStateFromSnapshot(offerState.localState)
+      onSnapshotRecovery(SnapshotOffer(offer.metadata, offerState.userData))
+    }
     case x:Any =>
       log.error(s"Ignoring msg of unknown type: ${x.getClass}")
   }
 
+
+  override def saveSnapshot(snapshot: Any): Unit = {
+    val actorState = StoredEnhancedPersistentActorState(
+      isProcessingEvent = isProcessingEvent,

```

Comment:
I think this approach will introduce bugs in the future.. If we add new variable to internal state, then we do not know that we have to add it here.

All of the internal state should probably be moved into a case class, so that storing the state would only be to store a single object.

Reply:
Ah sure, that is a possibility.

Reply:
I added a start.
I kept some state on the outside, and that was a bit of guesswork what to put where.
I added a method which only returns true if those methods are in their initial state. Keeping this in sync is not nice though, and prone for future bugs still :(